lib/srv/desktop/windows_server.go

```diff
@@ -858,6 +862,24 @@ func (s *WindowsService) connectRDP(ctx context.Context, log *slog.Logger, tdpCo
 	tdpConn.OnSend = s.makeTDPSendHandler(ctx, recorder, delay, tdpConn, audit)
 	tdpConn.OnRecv = s.makeTDPReceiveHandler(ctx, recorder, delay, tdpConn, audit)
 	width, height := desktop.GetScreenSize()
+
+	computerName, ok := desktop.GetLabel(types.DiscoveryLabelWindowsDNSHostName)
+	if !ok {
+		if computerName, err = utils.Host(desktop.GetAddr()); err != nil {
+			return trace.Wrap(err)
+		}
+		if len(net.ParseIP(computerName)) != 0 {
+			computerName = "missing.computer.name"
+		}
+	}
+
+	kdcAddr := s.cfg.KDCAddr
+	if kdcAddr == "" && s.cfg.LDAPConfig.Addr != "" {
+		if kdcAddr, err = utils.Host(s.cfg.LDAPConfig.Addr); err != nil {
+			return trace.Wrap(err)
```

Comment:
Add a message to the error. Something like

```
trace.Wrap(err, "KDC address is unspecified and LDAP address is invalid")
```

Also, should this be a fatal error? If we're connecting to a non-AD host we don't need to error out here, right?

Reply:
I've added check to only do this for AD desktops